### PR TITLE
Update to Pydantic v2

### PR DESCRIPTION
**Pull request by dpeachey**
Update all code to work with Pydantic v2

### CODE REVIEW

**Comment by dpeachey** [comment]
Relates to #14

File: examples/ietf-interfaces/restconf.py [hunk 1]
```diff
@@ -27,10 +27,8 @@
 
 
 print("#" * 17, "Convert received data into pydantic model", "#" * 17)
-from ietf_interfaces import Model, BaseConfig
-from pydantic import Extra
+from ietf_interfaces import Model
 
-BaseConfig.extra = Extra.ignore
 model = Model.parse_obj(response.json())
```
Comment (ubaumann) [inline examples/ietf-interfaces/restconf.py:32]: Use `model_validate`

File: examples/ietf-interfaces/restconf.py [hunk 2]
```diff
@@ -27,10 +27,8 @@
...
 
 print(model.json(exclude_defaults=True, by_alias=True, indent=2))
```
Comment (ubaumann) [inline examples/ietf-interfaces/restconf.py:34]: Use model_dump_json

File: examples/ietf-interfaces/restconf.py [hunk 3]
```diff
@@ -41,18 +39,18 @@
...
 from ietf_interfaces import EnabledLeaf, DescriptionLeaf, InterfacesContainer
 
 interface = model.interfaces.interface[1]
-interface.enabled = EnabledLeaf(__root__=True)
+interface.enabled = EnabledLeaf(root=True)
 interface.description = DescriptionLeaf(
-    __root__="https://pydantify.github.io/pydantify/"
+    root="https://pydantify.github.io/pydantify/"
 )
 print(interface.json(exclude_defaults=True, by_alias=True, indent=2))
```
Comment (ubaumann) [inline examples/ietf-interfaces/restconf.py:55]: Use model_dump_json

Comment (ubaumann) [reply examples/ietf-interfaces/restconf.py:55]: Also on lines 60, 66
Use `model_dump` on line 75

**Comment by ubaumann** [comment]
The line number references must be updated because the code changed in the example. We can merge it, and I will fix it afterwards, or you can quickly change it. What do you prefer?

docs/examples/hello_world.md
```diff
-python title="create_json.py" linenums="3"
---8<-- "examples/hello_world/create_json.py:3:10"
+python title="create_json.py" linenums="2"
+--8<-- "examples/hello_world/create_json.py:2:8"

-python title="create_json.py" linenums="15"
---8<-- "examples/hello_world/create_json.py:15:15"
+python title="create_json.py" linenums="13"
+--8<-- "examples/hello_world/create_json.py:13:13"

-python title="create_json.py" linenums="20"
---8<-- "examples/hello_world/create_json.py:20:25"
+python title="create_json.py" linenums="18"
+--8<-- "examples/hello_world/create_json.py:18:23"

-python title="create_json.py" linenums="30"
---8<-- "examples/hello_world/create_json.py:30:31"
+python title="create_json.py" linenums="28"
+--8<-- "examples/hello_world/create_json.py:28:29"

-python title="create_json.py" linenums="36"
---8<-- "examples/hello_world/create_json.py:36:39"
+python title="create_json.py" linenums="34"
+--8<-- "examples/hello_world/create_json.py:34:37"

-python title="create_json.py" linenums="43"
---8<-- "examples/hello_world/create_json.py:43:51"
+python title="create_json.py" linenums="41"
+--8<-- "examples/hello_world/create_json.py:41:49"
```

docs/examples/ietf_interfaces.md
```diff
 python title="restconf.py" linenums="30"
---8<-- "examples/ietf-interfaces/restconf.py:30:36"
+--8<-- "examples/ietf-interfaces/restconf.py:30:34"

-python title="restconf.py" linenums="41"
---8<-- "examples/ietf-interfaces/restconf.py:41:47"
+python title="restconf.py" linenums="39"
+--8<-- "examples/ietf-interfaces/restconf.py:39:45"

-python title="restconf.py" linenums="50"
---8<-- "examples/ietf-interfaces/restconf.py:50:57"
+python title="restconf.py" linenums="48"
+--8<-- "examples/ietf-interfaces/restconf.py:48:55"

-python title="restconf.py" linenums="61"
---8<-- "examples/ietf-interfaces/restconf.py:61:62"
+python title="restconf.py" linenums="59"
+--8<-- "examples/ietf-interfaces/restconf.py:59:60"

-python title="restconf.py" linenums="66"
---8<-- "examples/ietf-interfaces/restconf.py:66:69"
+python title="restconf.py" linenums="64"
+--8<-- "examples/ietf-interfaces/restconf.py:64:67"

-python title="restconf.py" linenums="74"
---8<-- "examples/ietf-interfaces/restconf.py:74:81"
+python title="restconf.py" linenums="72"
+--8<-- "examples/ietf-interfaces/restconf.py:72:79"
 ```

**Review by ubaumann — COMMENTED** [review]
Great work. Thank you very much. I just added some comments regarding documentation.

**Comment by dpeachey** [comment]
Thanks for the review! I'll make the suggested updates and push another commit.

**Review by ubaumann — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**